### DISCUSSION TITLE
How to run simulation in multicore?

### DISCUSSION THREAD

**Discussion by dipta005** [ProjectPhysX/FluidX3D#212] [Q&A]
I tried Openlb, there if I want to run simulations on multicore, I just to run: mpirun -np 4.... But, what is the method here? I couldn't understand the readme file.
_Thanks in advance.

**Comment by ProjectPhysX** [comment]
Hi @dipta005,

- Please follow the instructions [here](https://github.com/ProjectPhysX/FluidX3D/blob/master/DOCUMENTATION.md#0-install-gpu-drivers-and-opencl-runtime) to install the Intel CPU Runtime for OpenCL and/or PoCL.
- Run FluidX3D with `./make.sh 2` or `bin\FluidX3D 2`, where `2` is the OpenCL device ID of the CPU. This will run FluidX3D on all available CPU cores, using AVX2 or AVX-512 depending on what the CPU supports.
- If both Intel CPU Runtime and PoCL are installed, the CPU will show up twice as OpenCL device, once with the Intel CPU Runtime and once with PoCL. You can select between backends with the OpenCL device ID.
  ```
  |----------------.------------------------------------------------------------|
  | Device ID    0 | Intel(R) Core(TM) i7-8700K CPU @ 3.70GHz                   |
  | Device ID    1 | pthread-Intel(R) Core(TM) i7-8700K CPU @ 3.70GHz           |
  |----------------'------------------------------------------------------------|
  ```
- You can also run on only selected CPU cores with `numactl --localalloc --physcpubind=0-18,38-57 bin/FluidX3D 2`. The lists after physcpubind are the CPU threads and the 2 after bin/FluidX3D is the OpenCL device ID.

Kind regards,
Moritz

**Reply by dipta005** [reply]
I am using Amd Ryzen cpu.

**Reply by ProjectPhysX** [reply]
Both the Intel CPU Runtime for OpenCL and PoCL also work for AMD CPUs.
